inspect/api_deps.js

```diff
@@ -24,22 +26,148 @@ declare module 'grist' {
+        * a moment date. Option 'keepEncoded' skips the decoding step.
+        */
+    export function fetchSelectedTable(options?: {
+            keepEncoded?: boolean;
+    }): Promise<any>;
+    /**
+        * Fetches current selected record as for [[GristView.fetchSelectedRecord]],
+        * but decoding data by default, replacing e.g. ['D', timestamp] with
+        * a moment date. Option 'keepEncoded' skips the decoding step.
+        */
+    export function fetchSelectedRecord(rowId: number, options?: {
+            keepEncoded?: boolean;
+    }): Promise<any>;
+    /**
+        * A collection of methods for fetching document data. The
+        * fetchSelectedTable and fetchSelectedRecord methods are
+        * overridden to decode data by default.
+        */
     export const docApi: GristDocAPI & GristView;
     export const on: (event: string | symbol, listener: (...args: any[]) => void) => Rpc;
+    /**
+        * Shortcut for [[WidgetAPI.getOption]]
+        */
+    export const getOption: (key: string) => Promise<any>;
+    /**
+        * Shortcut for [[WidgetAPI.setOption]]
+        */
+    export const setOption: (key: string, value: any) => Promise<void>;
+    /**
+        * Shortcut for [[WidgetAPI.setOptions]]
+        */
+    export const setOptions: (options: {
+            [key: string]: any;
+    }) => Promise<void>;
+    /**
+        * Shortcut for [[WidgetAPI.getOptions]]
+        */
+    export const getOptions: () => Promise<object | null>;
+    /**
+        * Shortcut for [[WidgetAPI.clearOptions]]
+        */
+    export const clearOptions: () => Promise<void>;
+    /**
+        * Get access to a table in the document. If no tableId specified, this
+        * will use the current selected table (for custom widgets).
+        * If a table does not exist, there will be no error until an operation
+        * on the table is attempted.
+        */
+    export function getTable(tableId?: string): TableOperations;
+    /**
+        * Get an access token, for making API calls outside of the custom widget
+        * API. There is no caching of tokens. The returned token can
+        * be used to authorize regular REST API calls that access the content of the
+        * document. For example, in a custom widget for a table with a 'Photos' column
+        * containing attachments, the following code will update the 'src' of an
+        * image with id 'the_image' to show the attachment:
+        * '''js
+        * grist.onRecord(async (record) => {
+        *   const tokenInfo = await grist.docApi.getAccessToken({readOnly: true});
+        *   const img = document.getElementById('the_image');
+        *   const id = record.Photos[0];  // get an id of an attachment - there could be several
+        *                                 // in a cell, for this example we just take the first.
+        *   const src = '${tokenInfo.baseUrl}/attachments/${id}/download?auth=${tokenInfo.token}';
```

Comment:
I just noticed it. @arnaudpeich can you add a simple replace to the generate.js file (line 56) to escape these characters in the string? Something like this would be enough:

`${contents.replace(/\$/g, "\\$")}`